symengine/pow.cpp

```diff
@@ -15,49 +15,49 @@ namespace SymEngine {
 Pow::Pow(const RCP<const Basic> &base, const RCP<const Basic> &exp)
     : base_{base}, exp_{exp}
 {
-    SYMENGINE_ASSERT(is_canonical(base, exp))
+    SYMENGINE_ASSERT (base != null)
+    SYMENGINE_ASSERT (exp != null)
+    SYMENGINE_ASSERT(is_canonical(*base, *exp))
```

Comment:
I think you don't need to check `base != null`, because the `SYMENGINE_ASSERT` is only executed in Debug mode, and in Debug mode, `*base` will raise an exception if it is null. So you can remove the two `null` checks.